clients/src/main/java/org/apache/kafka/common/protocol/Protocol.java

```diff
@@ -693,11 +708,33 @@
 
     public static final Schema[] UPDATE_METADATA_REQUEST = new Schema[] {UPDATE_METADATA_REQUEST_V0, UPDATE_METADATA_REQUEST_V1};
     public static final Schema[] UPDATE_METADATA_RESPONSE = new Schema[] {UPDATE_METADATA_RESPONSE_V0, UPDATE_METADATA_RESPONSE_V1};
+    public static final short[] DEPRECATED_UPDATE_METADATA_REQUEST = new short[] {};
+
+    /* Protocol version api */
+    public static final Schema PROTOCOL_VERSION_REQUEST_V0 = new Schema();
```

Comment:
Would there be any benefit to taking a list of apiKeys that you care about instead of always reporting all? And an empty list means all. 

I am not sure about this. Just posing the question.